### PR TITLE
Open rosa

### PR DESCRIPTION
**Pull request by adammichaelwood**
addresses #9 

## What is included in this PR?

The OpenRosa intro and documentation of the five "core" OpenRosa APIs

## What is left to be done in the addressed issue?
There are a number of additional "unofficial" and proposed APIs in the OpenRosa docs.
They may need to be pulled in as well.

## What problems did you encounter?

 - The metadata schema took the longest, because I hadn't fully understood the XForm document structure.

### CODE REVIEW

File: openrosa-form-submission.rst [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+.. note::
+  Using digest authentication and https when communicating with a server does not require any redirects --- you can have authentication and secure transport without redirects.
+
+.. _content:
+
+Content
+~~~~~~~~~~
+
+OpenRosa submissions (and responses) MUST provide headers conforming to the OpenRosa Request standard.
```
Comment (lognaturel) [inline openrosa-form-submission.rst]: link to the page with that standard

File: openrosa-form-submission.rst [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+Successful server responses MUST include an ``X-OpenRosa-Accept-Content-Length`` header in addition to the :ref:`required OpenRosa Request response headers <openrosa-responses>`. The ``X-OpenRosa-Accept-Content-Length`` header specifies this server's recommended maximum size for a POST body, in bytes.
+
+Servers SHOULD include this header in their error responses. However, clients MUST NOT rely on the presence of this header (or any OpenRosa header) in every error response.
+
+.. warning::
```
Comment (lognaturel) [inline openrosa-form-submission.rst]: I don't know if this should be a warning. It's more of an example of a case in which a header wouldn't be provided.

Comment (adammichaelwood) [reply openrosa-form-submission.rst]: changed it to `note`

File: openrosa-http.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+.. code-block:: xml
+
+    <OpenRosaResponse xmlns="http://openrosa.org/http/response" items="3">
+        <message nature="submit_success">Some message to be displayed to the user</message>
+        <Registration xmlns="http://example.org/user/registration"/> <!-- optional -->
+        <!--... Some platform specific response ...-->
+    </OpenRosaResponse>
+    
+APIs using this response can then include their namespaced payload inside of the response. This response should be handled in a unified way on the client, regardless of the specific submission. That is, any time an OpenRosa response envelope is received, any of its payloads should be parsed properly.
```
[the remark this answers is not in the document]
Comment (adammichaelwood) [reply openrosa-http.rst]: I assumed it meant -- "Do the same thing every time."
But that seems... redundant at best.

[the remark this answers is not in the document]
Comment (adammichaelwood) [reply openrosa-http.rst]: I've changed it to:

> This response should be handled by the client the same way, regardless of the specific submission.

File: openrosa-metadata.rst [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+This document details the metadata scheme to be used in an XForm intended for use with OpenRosa compliant servers. This metadata is used to ensure that critical or extremely useful data pertaining to the XForms and submissions made by client devices is captured and stored.
+
+There are two types of metadata in a form submission. 
+
+ - Form metadata — Data about the identity and version of the XForm used to create the data being submitted. 
+ - Submission metadata — Data about the submission itself.
+
+.. _blank-form-metadata:
+ 
+Blank Form Metadata
```
[the remark this answers is not in the document]
Comment (adammichaelwood) [reply openrosa-metadata.rst]: On some level I feel that way about this entire document.

File: openrosa-metadata.rst [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+The value of ``version`` MUST be incremented when any part of the form changes. 
+
+If ``version`` is not present, it is handled as NULL. 
+
+.. _completed-form-metadata:
+
+Completed Form Metadata
+--------------------------
+
+In a completed form, metadata is provided in a ``<meta>`` element inside the ``<model>`` element.
```
[the remark this answers is not in the document]
Comment (adammichaelwood) [reply openrosa-metadata.rst]: But that is not what ODK Collect does currently.
I looked at completed forms out of Collect, and the meta data is in an element named `<meta>`

Also, this is as it was in the original document.

**Comment by adammichaelwood** [comment]
I've made the corrections.

I agree with your two comments about stuff that should be in the other doc, but on some level I feel that way about the entire metadata doc.

I think that might require a longer conversation in the future... and perhaps *all* the open rosa stuff and the XForm spec should be together in their own repo...

For the mean time, I think we should merge it in as is.

**Comment by adammichaelwood** [comment]
@lognaturel I made (most of) the requested changes. Can you review and (I hope) merge?

**Pull request merged**